packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2158,7 +2256,15 @@ export class ContainerRuntime
 		// Do shallow copy of message, as the processing flow will modify it.
 		const messageCopy = { ...messageArg };
 		for (const message of this.remoteMessageProcessor.process(messageCopy)) {
-			this.processCore(message, local, modernRuntimeMessage);
+			if (modernRuntimeMessage) {
+				this.processCore({
+					message: message as InboundSequencedContainerRuntimeMessage,
```

Comment:
```suggestion
					// Cast it since we expect it to be this based on modernRuntimeMessage computation above
					message: message as InboundSequencedContainerRuntimeMessage,
```